speculator/train_speculator.py

```diff
@@ -0,0 +1,321 @@
+import math
+import os
+import time
+
+import fire
+import torch
+import torch.optim as optim
+from fms.models import get_model
+from fms.models.llama import LLaMABlock
+from fms.utils import generation, tokenizers
+from fms_extras.models.speculator import MLPSpeculator
+from torch import distributed as dist
+from torch.distributed.fsdp import FullyShardedDataParallel as FSDP
+from torch.distributed.fsdp import ShardingStrategy
+from torch.optim.lr_scheduler import LambdaLR
+
+from fms_fsdp import config
+from fms_fsdp.utils.checkpointing_utils import Checkpointer
+from fms_fsdp.utils.config_utils import update_config
+from fms_fsdp.utils.dataloader_utils import get_data_loader, get_dummy_loader
+from fms_fsdp.utils.train_utils import (
+    get_policies,
+    get_profiler,
+    setup,
+    setup_environ_flags,
+)
+from speculator.train_speculator_utils import train_speculator
+
+
+os.environ["PYTORCH_CUDA_ALLOC_CONF"] = "expandable_segments:True"
+os.environ["TOKENIZERS_PARALLELISM"] = "false"
+
+
+def test_model(rank, model, arch, cfg, prompt_type="chat"):
```

Comment:
at some point it would be good to have an assertion added to this such that the training will not start without this passing

Reply:
What would you check in the assert?

Reply:
You could have a sample input/output file that the output should match the expected output